Modix/Startup.cs

```diff
@@ -61,9 +61,13 @@ public void ConfigureServices(IServiceCollection services)
             services.AddTransient<IConfigureOptions<StaticFileOptions>, StaticFilesConfiguration>();
             services.AddTransient<IStartupFilter, ModixConfigValidator>();
 
-            services.AddDbContext<ModixContext>(options => options
-                .UseNpgsql(_configuration.GetValue<string>(nameof(ModixConfig.DbConnection)), npgsqlOptions => npgsqlOptions
-                    .UseDateTimeOffsetTranslations()));
+            services
+                .AddDbContext<ModixContext>(options => options
+                    .UseNpgsql(_configuration.GetValue<string>(nameof(ModixConfig.DbConnection)), npgsqlOptions => npgsqlOptions
+                        .UseDateTimeOffsetTranslations()))
+                .AddSingleton<ModixContextAutoMigrationStartupAction>()
+                .AddSingleton<IBehavior>(serviceProvider => serviceProvider.GetRequiredService<ModixContextAutoMigrationStartupAction>())
+                .AddSingleton<IModixContextAutoMigrationStartupAction>(serviceProvider => serviceProvider.GetRequiredService<ModixContextAutoMigrationStartupAction>());
```

Comment:
These should use the form `.AddSingleton<IBehavior, ModixContextAutoMigrationStartupAction>()` if they don't need to do any special resolution logic

Reply:
They do need special resolution logic. This logic. The logic that ensures that the same instance is used for both registrations.

Reply:
Oh, I understand now, thanks